### ISSUE TITLE
Can't submit existing form causes Invalid value for 'encryptedTicket'

### ISSUE THREAD

**Issue by GianniDPC** [umbraco/Umbraco.Forms.Issues#632]
Reproduction
------------

### Bug summary

The problem occurs when trying to submit an existing form with files linked to them. I have upgraded to version 8.7.6 to apply the security patch. I also edited the appropriate  FileUpload razor files to reflect the changes.

I don't quite understand why this is happening. I have to add that I created my own 'myFileUploader' type but the only thing it does is inherit from FileUpload.

### Specifics

I am using Umbraco Forms version 8.7.6 together with umbraco 8.5.3

### Steps to reproduce

See summary

### Expected result

The form to submit and persist the changes.

### Actual result

Following exception occurs:

```
2021-08-25 08:43:28.8684 | System.ArgumentException: Invalid value for 'encryptedTicket' parameter.
   at System.Web.Security.FormsAuthentication.Decrypt(String encryptedTicket)
   at Umbraco.Core.StringExtensions.DecryptWithMachineKey(String value) in d:\a\1\s\src\Umbraco.Core\StringExtensions.cs:line 309
   at Umbraco.Forms.Core.Providers.FieldTypes.FileUpload.DecryptFilePath(String path)
   at Umbraco.Forms.Core.Providers.FieldTypes.FileUpload.ValidateField(Form form, Field field, IEnumerable`1 postedValues, HttpContextBase context, IFormStorage formStorage)
   at Umbraco.Forms.Web.Controllers.UmbracoFormsController.ValidateFormState(FormViewModel model, Form form, HttpContextBase context)
   at Umbraco.Forms.Web.Controllers.UmbracoFormsController.HandleForm(FormViewModel model)
   at lambda_method(Closure , ControllerBase , Object[] )
   at System.Web.Mvc.ActionMethodDispatcher.Execute(ControllerBase controller, Object[] parameters)
   at System.Web.Mvc.ReflectedActionDescriptor.Execute(ControllerContext controllerContext, IDictionary`2 parameters)
   at System.Web.Mvc.ControllerActionInvoker.InvokeActionMethod(ControllerContext controllerContext, ActionDescriptor actionDescriptor, IDictionary`2 parameters)
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c.<BeginInvokeSynchronousActionMethod>b__9_0(IAsyncResult asyncResult, ActionInvocation innerInvokeState)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResult`2.CallEndDelegate(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResultBase`1.End()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.EndInvokeActionMethod(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.AsyncInvocationWithFilters.<>c__DisplayClass11_0.<InvokeActionMethodFilterAsynchronouslyRecursive>b__0()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.AsyncInvocationWithFilters.<>c__DisplayClass11_2.<InvokeActionMethodFilterAsynchronouslyRecursive>b__2()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.AsyncInvocationWithFilters.<>c__DisplayClass11_2.<InvokeActionMethodFilterAsynchronouslyRecursive>b__2()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.AsyncInvocationWithFilters.<>c__DisplayClass11_2.<InvokeActionMethodFilterAsynchronouslyRecursive>b__2()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.AsyncInvocationWithFilters.<>c__DisplayClass11_2.<InvokeActionMethodFilterAsynchronouslyRecursive>b__2()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass7_0.<BeginInvokeActionMethodWithFilters>b__1(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResult`1.CallEndDelegate(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResultBase`1.End()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.EndInvokeActionMethodWithFilters(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass3_6.<BeginInvokeAction>b__4()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass3_1.<BeginInvokeAction>b__1(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResult`1.CallEndDelegate(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResultBase`1.End()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.EndInvokeAction(IAsyncResult asyncResult)
   at System.Web.Mvc.Controller.<>c.<BeginExecuteCore>b__152_1(IAsyncResult asyncResult, ExecuteCoreState innerState)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncVoid`1.CallEndDelegate(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResultBase`1.End()
   at System.Web.Mvc.Controller.EndExecuteCore(IAsyncResult asyncResult)
   at System.Web.Mvc.Controller.<>c.<BeginExecute>b__151_2(IAsyncResult asyncResult, Controller controller)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncVoid`1.CallEndDelegate(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResultBase`1.End()
   at System.Web.Mvc.Controller.EndExecute(IAsyncResult asyncResult)
   at System.Web.Mvc.Controller.System.Web.Mvc.Async.IAsyncController.EndExecute(IAsyncResult asyncResult)
   at System.Web.Mvc.MvcHandler.<>c.<BeginProcessRequest>b__20_1(IAsyncResult asyncResult, ProcessRequestState innerState)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncVoid`1.CallEndDelegate(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResultBase`1.End()
   at System.Web.Mvc.MvcHandler.EndProcessRequest(IAsyncResult asyncResult)
   at System.Web.Mvc.MvcHandler.System.Web.IHttpAsyncHandler.EndProcessRequest(IAsyncResult result)
   at System.Web.HttpApplication.CallHandlerExecutionStep.System.Web.HttpApplication.IExecutionStep.Execute()
   at System.Web.HttpApplication.ExecuteStepImpl(IExecutionStep step)
   at System.Web.HttpApplication.ExecuteStep(IExecutionStep step, Boolean& completedSynchronously) | System.ArgumentException: Invalid value for 'encryptedTicket' parameter.
   at System.Web.Security.FormsAuthentication.Decrypt(String encryptedTicket)
   at Umbraco.Core.StringExtensions.DecryptWithMachineKey(String value) in d:\a\1\s\src\Umbraco.Core\StringExtensions.cs:line 309
   at Umbraco.Forms.Core.Providers.FieldTypes.FileUpload.DecryptFilePath(String path)
   at Umbraco.Forms.Core.Providers.FieldTypes.FileUpload.ValidateField(Form form, Field field, IEnumerable`1 postedValues, HttpContextBase context, IFormStorage formStorage)
   at Umbraco.Forms.Web.Controllers.UmbracoFormsController.ValidateFormState(FormViewModel model, Form form, HttpContextBase context)
   at Umbraco.Forms.Web.Controllers.UmbracoFormsController.HandleForm(FormViewModel model)
   at lambda_method(Closure , ControllerBase , Object[] )
   at System.Web.Mvc.ActionMethodDispatcher.Execute(ControllerBase controller, Object[] parameters)
   at System.Web.Mvc.ReflectedActionDescriptor.Execute(ControllerContext controllerContext, IDictionary`2 parameters)
   at System.Web.Mvc.ControllerActionInvoker.InvokeActionMethod(ControllerContext controllerContext, ActionDescriptor actionDescriptor, IDictionary`2 parameters)
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c.<BeginInvokeSynchronousActionMethod>b__9_0(IAsyncResult asyncResult, ActionInvocation innerInvokeState)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResult`2.CallEndDelegate(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResultBase`1.End()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.EndInvokeActionMethod(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.AsyncInvocationWithFilters.<>c__DisplayClass11_0.<InvokeActionMethodFilterAsynchronouslyRecursive>b__0()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.AsyncInvocationWithFilters.<>c__DisplayClass11_2.<InvokeActionMethodFilterAsynchronouslyRecursive>b__2()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.AsyncInvocationWithFilters.<>c__DisplayClass11_2.<InvokeActionMethodFilterAsynchronouslyRecursive>b__2()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.AsyncInvocationWithFilters.<>c__DisplayClass11_2.<InvokeActionMethodFilterAsynchronouslyRecursive>b__2()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.AsyncInvocationWithFilters.<>c__DisplayClass11_2.<InvokeActionMethodFilterAsynchronouslyRecursive>b__2()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass7_0.<BeginInvokeActionMethodWithFilters>b__1(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResult`1.CallEndDelegate(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResultBase`1.End()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.EndInvokeActionMethodWithFilters(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass3_6.<BeginInvokeAction>b__4()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass3_1.<BeginInvokeAction>b__1(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResult`1.CallEndDelegate(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResultBase`1.End()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.EndInvokeAction(IAsyncResult asyncResult)
   at System.Web.Mvc.Controller.<>c.<BeginExecuteCore>b__152_1(IAsyncResult asyncResult, ExecuteCoreState innerState)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncVoid`1.CallEndDelegate(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResultBase`1.End()
   at System.Web.Mvc.Controller.EndExecuteCore(IAsyncResult asyncResult)
   at System.Web.Mvc.Controller.<>c.<BeginExecute>b__151_2(IAsyncResult asyncResult, Controller controller)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncVoid`1.CallEndDelegate(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResultBase`1.End()
   at System.Web.Mvc.Controller.EndExecute(IAsyncResult asyncResult)
   at System.Web.Mvc.Controller.System.Web.Mvc.Async.IAsyncController.EndExecute(IAsyncResult asyncResult)
   at System.Web.Mvc.MvcHandler.<>c.<BeginProcessRequest>b__20_1(IAsyncResult asyncResult, ProcessRequestState innerState)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncVoid`1.CallEndDelegate(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResultBase`1.End()
   at System.Web.Mvc.MvcHandler.EndProcessRequest(IAsyncResult asyncResult)
   at System.Web.Mvc.MvcHandler.System.Web.IHttpAsyncHandler.EndProcessRequest(IAsyncResult result)
   at System.Web.HttpApplication.CallHandlerExecutionStep.System.Web.HttpApplication.IExecutionStep.Execute()
   at System.Web.HttpApplication.ExecuteStepImpl(IExecutionStep step)
   at System.Web.HttpApplication.ExecuteStep(IExecutionStep step, Boolean& completedSynchronously)
```

**Comment by kevinlefebure** [comment]
Having the same issue with Forms v8.3.4 and Umbraco 8.5.1.

**Comment by kevinlefebure** [comment]
Any progress on this issue? Currently this is blocking normal operation on a client's project...
And reverting the patch is not a fix of course.

**Comment by AndyButland** [comment]
Yes, I've got a fix for this in the PR linked to this ticket - which will be in 8.8.0.  That's not out for a couple of weeks though.  In the meantime, I can prepare a dll for download to update 8.7.6.

**Comment by AndyButland** [comment]
The dll can be downloaded from [here](https://drive.google.com/file/d/1lieMSa765yDDwlrhq-_A93wFTXo_9PD0/view?usp=sharing).  This is a patch for 8.7.6 so if running that version, you should be able to replace the single dll provided in the linked zip file, in order to verify and run with the fix.

As noted this will be included in 8.8.0, expected in a couple of weeks.

**Comment by kevinlefebure** [comment]
@AndyButland Thanks for that, only problem is, I'm running v8.3.4... 
Is it possible to do the same thing for this version? Updating is something that I don't like doing with live projects... (as proven with this patch ;-) )
Will there be a fix for each version of this patch?

**Comment by AndyButland** [comment]
I'd encourage if you can to upgrade to the latest minor for Forms - as I expect you'd benefit form quite a few bug fixes and features that have been introduced since the version you are on. It's not normally the process to back-port bug fixes to previous minors, so no, I wasn't intending to provide a patch for them all - just the latest.  We did this for the security patch as of course it's important we provide the easiest path we can for customers to get onto a version with the issue resolved.

That said, I do appreciate that this bug, even if only manifesting in certain scenarios - where forms are being re-edited, and with file uploads - has been introduced as part of the security patch, so I'll have a look at creating a similar dll patch for 8.3.  Hopefully that'll tide you over while you try to find some time to upgrade to the latest version of Forms and then can be back on the official release versions.

**Comment by AndyButland** [comment]
Here's the [single dll patch for 8.3.4](https://drive.google.com/file/d/1LzDUj4uaSLsYUJdhcdCOSqSUc1w_cs3k/view?usp=sharing),

**Comment by elit0451** [comment]
The issue has been fixed and is due in the next release 🎊

**Comment by AndyButland** [comment]
@NurhakKaya - I'll prepare a back-port of this for this for the latest version of Forms for Umbraco 7, which will be released in the 7.5.8 of Forms.  Once that's out, would you be able to upgrade to that?  Seems that should be OK as the minor release from 7.4 to 7.5 was designated a minor due to a new feature being introduced, and there were no documented breaking changes.

**Comment by NurhakKaya** [comment]
Thanks @AndyButland , that'd ve very good. When do you plan to release v7.5.8 of Forms, please?

**Comment by NurhakKaya** [comment]
A pre-release would be great if you can Andy. Thank you for your help.

**Comment by AndyButland** [comment]
Forms 7.5.8 with this fix ported back is available now @NurhakKaya on NuGet and our.  It'll be rolled out as an auto-upgrade on Umbraco Cloud projects next week.

**Comment by NurhakKaya** [comment]
Excellent work @AndyButland , thank you so much for sorting this out so quickly. #h5yr

**Comment by NurhakKaya** [comment]
I can confirm that @AndyButland 's fix works with Umbraco Forms v7.5.8. Thanks again for your help Andy.

**Comment by AndyButland** [comment]
You're welcome, and thanks for confirming.
